### ISSUE TITLE
Erronous "unnecessary use of `to_string`" with `colored::ColoredString`

### ISSUE THREAD

**Issue by stevecheckoway** [rust-lang/rust-clippy#11102]
### Summary

Clippy suggests replacing `&foo.to_string()` with `&foo`. When `foo` is a `ColoredString` from the popular `colored` crate, this suggestion is incorrect because `&foo` dereferences to just the underlying characters without including the terminal control characters for setting the colors. 

### Lint Name

clippy::unnecessary_to_owned

### Reproducer

I tried this code:

```rust
use colored::{ColoredString, Colorize};

fn main() {
    let mut s = String::new();
    let c: ColoredString = "red".red();
    s.push_str(&c);
    s.push('\n');
    s.push_str(&c.to_string());
    println!("{s}");
}
```

I saw this happen:

Clippy warns on the `s.push_str(&c.to_string())`. The code prints `red` twice but only the second is colored red.

I expected to see this happen:

I expected no warning from clippy because the `.to_string()` is correct (and seems required).

### Version

```text
rustc 1.70.0 (90c541806 2023-05-31)
binary: rustc
commit-hash: 90c541806f23a127002de5b4038be731ba1458ca
commit-date: 2023-05-31
host: aarch64-apple-darwin
release: 1.70.0
LLVM version: 16.0.2
```

**Comment by y21** [comment]
This looks like #10033, which was fixed in #10913 and should no longer lint on nightly. I think the fix will be in the next stable release if I'm not mistaken

**Comment by stevecheckoway** [comment]
@y21 Thank you. You are correct that this doesn't warn on nightly.
